lib/route.js

```diff
@@ -468,7 +468,7 @@ function buildRouting (options) {
 
     const request = new context.Request(id, params, req, query, childLogger, context)
     const reply = new context.Reply(res, request, childLogger)
-
+    request.port = Number(request.host && request.host.split(':').length > 1 ? request.host.split(':')[1] : null)
```

Comment:
This can be inferred directly from the `headers` when requested. This might not be really needed here 🤔

Reply:
But in case of a proxy, the port might not be there. So how can I infer the port?

Reply:
Have into account that this information is already available for the `Request` object when it's being instantiated. For instance, if `trustProxy` is set, you can infer it with the `host` property, see how it works [here](https://github.com/fastify/fastify/pull/4766/files/66f29d902ab80d1888a623c6ffa7814c6c7e620d#diff-2871801a54601791d9e1004f04b2df5d7adc881560285ff3fa6bee7bf9697857R118)